src/main/java/ch/njol/skript/expressions/ExprScript.java

```diff
@@ -18,76 +18,102 @@
  */
 package ch.njol.skript.expressions;
 
+import ch.njol.skript.ScriptLoader;
 import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptCommand;
 import ch.njol.skript.doc.Description;
-import ch.njol.skript.doc.Events;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.parser.ParserInstance;
+import org.jetbrains.annotations.Nullable;
 import org.skriptlang.skript.lang.script.Script;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
 import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
 
-@Name("Script Name")
-@Description("Holds the current script's name (the file name without '.sk').")
+import java.util.Arrays;
+import java.util.Objects;
+
+@Name("Script")
+@Description("The current script, all loaded scripts, or a script from its (file) name.")
 @Examples({
+	"on script load:",
+	"\tbroadcast \"Loaded %the current script%\"",
 	"on script load:",
 	"\tset {running::%script%} to true",
 	"on script unload:",
-	"\tset {running::%script%} to false"
+	"\tset {running::%script%} to false",
+	"set {script} to the script named \"weather.sk\""
 })
 @Since("2.0")
-@Events("Script Load/Unload")
-public class ExprScript extends SimpleExpression<String> {
-	
+public class ExprScript extends SimpleExpression<Script> {
+
 	static {
-		Skript.registerExpression(ExprScript.class, String.class, ExpressionType.SIMPLE,
-			"[the] script[['s] name]",
-			"name of [the] script"
+		Skript.registerExpression(ExprScript.class, Script.class, ExpressionType.SIMPLE,
+			"[the] [current] script",
+			"[the] script[s] [named] %strings%",
+			"all [loaded] scripts"
```

Comment:
already exists as ExprScripts, but we'll have to figure out what to do about the return types